### PR TITLE
refactor: Browser sort order + direction

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
* more changes from https://github.com/ankidroid/Anki-Android/pull/14843
* merging (not reviewing) is blocked on https://github.com/ankidroid/Anki-Android/pull/14897

## Approach
Extract method, add helper enum for return value, inline properties and use ViewModel in Dialog

## How Has This Been Tested?
Tests & manually

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 1]
```diff
@@ -164,12 +165,10 @@ open class CardBrowser :
     private var mCurrentCardId
         get() = viewModel.currentCardId
         set(value) { viewModel.currentCardId = value }
-    private var mOrder
+    private val mOrder
         get() = viewModel.order
-        set(value) { viewModel.order = value }
-    private var mOrderAsc
+    private val mOrderAsc
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:170]: Should have been removed in the inline operation

File: AnkiDroid/src/main/java/com/ichi2/anki/browser/CardBrowserViewModel.kt [hunk 2]
```diff
@@ -85,7 +92,20 @@ class CardBrowserViewModel(
...
             val cardsOrNotes = withCol { CardsOrNotes.fromCollection(this) }
             cardsOrNotesFlow.update { cardsOrNotes }
+
+            withCol {
+                sortTypeFlow.update { SortType.fromCol(config, sharedPrefs()) }
+                reverseDirectionFlow.update { ReverseDirection.fromConfig(config) }
+            }
         }
+
+        reverseDirectionFlow
```
[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/browser/CardBrowserViewModel.kt]: It would be foolish to argue. Fixed conflicts + ready to go

**Review by lukstbit — APPROVED** [review]
I think this is ok.

**Review by mikehardy — APPROVED** [review]
good to go

**Pull request merged**